src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -239,10 +242,10 @@ public Task StartHostAsync(RHostStartupInfo startupInfo, IRSessionCallback callb
             TaskUtilities.AssertIsOnBackgroundThread();
 
             _callback = callback;
-            _startupInfo = startupInfo;
+            _startupInfo = startupInfo ?? new RHostStartupInfo();
             RHost host;
             try {
-                var connectionInfo = new BrokerConnectionInfo(startupInfo.Name, this, startupInfo.RHostCommandLineArguments, timeout);
+                var connectionInfo = new BrokerConnectionInfo(Name, this, startupInfo.UseRHostCommandLineArguments, timeout);
```

Comment:
So why empty/null command line args are insufficient and we need a separate flag?